source/Xunit.Gherkin.Quick/Feature.cs

```diff
@@ -50,21 +50,34 @@ internal void Scenario(string scenarioName)
                     throw new Exception($"Step method partially matched but not selected. Step `{parsedStep.Text.Trim()}`, Method pattern `{matchingStepMethod.keywordAttribute.Pattern}`.");
 
                 var methodParams = matchingStepMethod.method.GetParameters();
-                object[] methodParamValues = null;
-                if (methodParams.Length > 0)
+
+                var methodParamsCount = (methodParams.LastOrDefault()?.ParameterType == typeof(DataTable))
```

Comment:
Why assume datatable will be only last? Is there any limitation in the Gherkin?
I would think it can be not only last, but any of them.

In other words, you need to (in my opinion):
- iterate through params.
- if param is datatable, then take the parsed table data and pass to it.
- if param is something else, then take the value as before.

Obviously, calculating the index of the parsed param will be little bit trickier, especially if the datatable is not the last one. But that is what devs expect I guess.

Reply:
Good point. I'll split this out a bit. I've got some unit tests lined up for this, but I'll submit that after the refactor PR goes through to keep the merging simple